### PR TITLE
KAFKA-12702: Fix NPE in networkListeners from BrokerServer in 2.8

### PR DESCRIPTION
**Pull request by wenbingshen**
Link to [KAFKA-12702](https://issues.apache.org/jira/browse/KAFKA-12702) , [10575](https://github.com/apache/kafka/pull/10575)

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by wenbingshen** [comment]
Dear @chia7712 , I request apply this pr to the 2.8 branch.

File: core/src/main/scala/kafka/server/BrokerServer.scala [hunk 1]
```diff
@@ -272,7 +273,7 @@ class BrokerServer(
       val networkListeners = new ListenerCollection()
       config.advertisedListeners.foreach { ep =>
         networkListeners.add(new Listener().
-          setHost(ep.host).
+          setHost(if (Utils.isBlank(ep.host)) InetAddress.getLocalHost.getCanonicalHostName else ep.host).
```
Comment (chia7712) [inline core/src/main/scala/kafka/server/BrokerServer.scala]: I feel it is ok to check blank manually. It won't bring huge code conflicts.

**Review by chia7712 — COMMENTED** [review]
@wenbingshen thanks for your contribution :)

File: core/src/main/scala/kafka/server/BrokerServer.scala (hunk 1 above)
Comment (wenbingshen) [reply core/src/main/scala/kafka/server/BrokerServer.scala]: > I feel it is ok to check blank manually. It won't bring huge code conflicts.

Good suggestion.Submit your feedback. :)

**Comment by chia7712** [comment]
@wenbingshen thanks for this backport!

**Pull request merged**